docs/migration-guide-NIO1-to-NIO2.md

```diff
@@ -0,0 +1,71 @@
+# NIO 1 to NIO 2 migration guide
+
+This migration guide is our recommendation to migrate from NIO 1 to NIO 2. None of the steps apart from changing your `Package.swift` and fixing the errors is actually required but this might help you doing the migration as fast as possible. For the NIO repositories and some example projects, we had really good success in migrating them fairly pain-free.
```

Comment:
Nit: s/this might help you doing/this might help you complete/

Reply:
Nit: s/we had really good success in migrating them fairly pain-free/we found that following these steps made migration fairly pain-free/.